airbyte-bootloader/src/main/java/io/airbyte/bootloader/SecretMigrator.java

```diff
@@ -55,34 +54,39 @@ static class ConnectorConfiguration {
    * Then for all the secret that are stored in a plain text format, it will save the plain text in
    * the secret manager and store the coordinate in the config DB.
    */
-  public void migrateSecrets() throws JsonValidationException, IOException {
+  public void migrateSecrets() throws JsonValidationException, IOException, ConfigNotFoundException {
     if (secretPersistence.isEmpty()) {
       log.info("No secret persistence is provided, the migration won't be run ");
 
       return;
     }
-    final List<StandardSourceDefinition> standardSourceDefinitions =
-        configPersistence.listConfigs(ConfigSchema.STANDARD_SOURCE_DEFINITION, StandardSourceDefinition.class);
+    final List<StandardSourceDefinition> standardSourceDefinitions = configRepository.listStandardSourceDefinitions(true);
 
-    final Map<UUID, JsonNode> definitionIdToSourceSpecs = standardSourceDefinitions
-        .stream().collect(Collectors.toMap(StandardSourceDefinition::getSourceDefinitionId,
-            def -> def.getSpec().getConnectionSpecification()));
+    final Map<UUID, ConnectorSpecification> definitionIdToSourceSpecs = standardSourceDefinitions
+        .stream().collect(Collectors.toMap(StandardSourceDefinition::getSourceDefinitionId, StandardSourceDefinition::getSpec));
 
-    final List<SourceConnection> sources = configPersistence.listConfigs(ConfigSchema.SOURCE_CONNECTION, SourceConnection.class);
+    final List<SourceConnection> sourcesWithoutSecrets = configRepository.listSourceConnection();
+    final List<SourceConnection> sourcesWithSecrets = new ArrayList<>();
```

Comment:
nit: This could be a map
```
final List<SourceConnection> sourcesWithSecrets = sourcesWithoutSecrets.map(source -> secretsReader.getSourceConnectionWithSecrets(source.getSourceId()))
```

Reply:
it throws a checked exception, which is why i didn't put it in the lambda.